enclave/core/tracee.c

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/bits/safemath.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/enclavelibc.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/sgxtypes.h>
+#include <openenclave/internal/trace.h>
+#include <openenclave/internal/utils.h>
+
+static oe_log_filter_t g_active_log_filter = {0};
+static char g_logging_filename[MAX_FILENAME_LEN];
+static bool g_debug_allowed_enclave = false;
+
+const char* get_filename_from_path(const char* path)
+{
+    if (path)
+    {
+        for (size_t i = oe_strlen(path) - 1; i >= 0; i--)
```

Comment:
This loop is not safe. If path does not have a '/' i will wrap around.

Reply:
added error condition checking.

Reply:
Out of curiosity, what are the compiler flags used for this file? oecore should be compiled with settings where this coding pattern is flagged as an error.

Reply:
Just checked with different compilers. It looks this type of warning was supposed to be improved in clang version 6, but somehow it cannot be triggered for this example. And in the case of GCC this warning seems to require Wextra to trigger. MSVC requires the Wall setting to trigger it. Unfortunately each compiler type identifies it differently so it is not simple to just add a single flag to CMake.